domain/src/ValueObjects/Transaction.php

```diff
@@ -75,28 +84,65 @@ public function isTransfer(): bool
         return $this->operation === Operation::Transfer;
     }
 
+    public function involvesNfts(): bool
+    {
+        return $this->sentAssetIsNft || $this->receivedAssetIsNft;
+    }
+
+    public function involvesSharePooling(): bool
+    {
+        return ! $this->sentAssetIsNft || ! $this->receivedAssetIsNft;
+    }
+
+    public function hasNetworkFee(): bool
+    {
+        return $this->networkFeeMarketValue?->isGreaterThan('0') ?? false;
+    }
+
+    public function networkFeeIsFiat(): bool
+    {
+        return $this->networkFeeCurrency ? FiatCurrency::tryFrom($this->networkFeeCurrency) !== null : false;
```

Comment:
ℹ️ There's a slight collision risk here, if the symbol of a cryptocurrency happens to match that of a fiat currency, or because of a typo when reporting the former